support/ci/shared.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -euo pipefail
+
+get_current_toolchain() {
+  # It turns out that every nightly version of rustfmt has slight tweaks from the previous version.
+  # This means that if we're always using the latest version, then we're going to have enormous
+  # churn. Even PRs that don't touch rust code will likely fail CI, since master will have been
+  # formatted with a different version than is running in CI. Because of this, we're going to pin
+  # the version of nightly that's used to run rustfmt and bump it when we do a new release.
```

Comment:
Maybe rehashing previous conversations - this looks like one more hoop to jump through every hab release, and create extra work downstream every time we release/re-pin nightly. Are we sure we are getting sufficient value from nightly rustfmt to bother?

Reply:
Yes

Reply:
The actual work involved is pretty small and the value (imo) is pretty great.

Reply:
As more of the configuration options in `rustfmt` stabilize, it may be worth revisiting this decision, but right now there are so many useful options that I think it's worth the extra effort to use nightly.